src/Weasel.Postgresql/Tables/Indexes/FullTextIndexDefinition.cs

```diff
@@ -0,0 +1,74 @@
+using JasperFx.Core;
+
+namespace Weasel.Postgresql.Tables.Indexes;
+
+public class FullTextIndexDefinition: IndexDefinition
+{
+    public const string DefaultRegConfig = "english";
+    public const string DataDocumentConfig = "data";
+
+    private readonly PostgresqlObjectName table;
+    private readonly string? indexName;
+    private readonly string? indexPrefix;
+
+    private string regConfig;
+
+    public FullTextIndexDefinition(
+        PostgresqlObjectName tableName,
+        string documentConfig,
+        string? regConfig = null,
+        string? indexName = null,
+        string? indexPrefix = null)
+    {
+        table = tableName;
+        this.regConfig = regConfig ?? DefaultRegConfig;
+        DocumentConfig = documentConfig;
+        this.indexName = indexName;
+        this.indexPrefix = indexPrefix;
+
+        Method = IndexMethod.gin;
+    }
+
+    public string? RegConfig
+    {
+        get => regConfig;
+        set => regConfig = value ?? DefaultRegConfig;
+    }
+
+    public string DocumentConfig { get; set; }
```

Comment:
I do understand, you are initializing it in ctor, but still, for the setter, wouldn't you check the value for null and set it to `DataDocumentConfig`?

Reply:
If we're using nullable reference types, the compiler will tell us we should not provide a null value there. If a user tries to cheat and forcibly put null, it's their mistake.

I'd like to get rid of `DataDocumentConfig` as it was my mistake to name it like that. Postgres names it `document` and if we put it into Weasel, then I think that we should keep naming generic.

I left `DataDocumentConfig` to not break people code when they upgrade Marten. 

Thoughts?

Reply:
Yeah, make sense!